### PR TITLE
Remove old browser support.

### PR DESCRIPTION
**Pull request by jdalton**
This PR removes old IE support (IE < 9), old Opera support, and old Safari support.
This also removes support for the `attachEvent` fork so IE 9 using the older branch is gone.

My editor removed trailing whitespace which is a bit noisy but also a nice thing.
I can try to parse it out if necessary.

I left the `Function#bind` shim in because of PhantomJS 1.9 (isn't a problem in 2.0).
I've test this in IE9 unit tests + functional tests and PhantomJS. I'll test in other browsers soon and report back.

Related to #306.

### CODE REVIEW

**Comment by jdalton** [comment]
Related to #306.

**Comment by jdalton** [comment]
Tested in IE 9, IE 11, Edge, Firefox (41 and 42), Safari (8 and 9), and Chrome (45 and 46).

File: src/prototype/dom/layout.js [hunk 1]
```diff
@@ -1677,27 +1530,6 @@
      !Element.descendantOf(element, document.body);
   }
 
-  // If the browser supports the nonstandard `getBoundingClientRect`
-  // (currently only IE and Firefox), it becomes far easier to obtain
-  // true offsets.
-  if ('getBoundingClientRect' in document.documentElement) {
```
Comment (savetheclocktower) [inline src/prototype/dom/layout.js]: Does this really need to go? If anything I figure more browsers are supporting `getBoundingClientRect` these days.

File: vendor/legacy_selector/repository/legacy_selector.js [hunk 2]
```diff
@@ -8,66 +8,17 @@ Prototype.LegacySelector = Class.create({
...
-      if ((/(\[[\w-]*?:|:checked)/).test(e))
-        return false;
-
-      if (IS_DESCENDANT_SELECTOR_BUGGY) return false;
-
-      return true;
-    }
-
-  })(),
+  shouldUseXPath: false,
```
Comment (savetheclocktower) [inline vendor/legacy_selector/repository/legacy_selector.js]: May as well remove this property entirely.

File: src/prototype/dom/layout.js (hunk 1 above)
Comment (jdalton) [inline src/prototype/dom/layout.js]: It's hard to tell from the diff but it's the only `viewportOffset` function I use (so not getting rid of it), it's the one used.

File: vendor/legacy_selector/repository/legacy_selector.js (hunk 2 above)
Comment (jdalton) [inline vendor/legacy_selector/repository/legacy_selector.js]: I wasn't sure on what to remove or keep but sure if you want.

Comment (jdalton) [inline vendor/legacy_selector/repository/legacy_selector.js]: Updated with it removed.

**Comment by savetheclocktower** [comment]
What will this do to, e.g., IE 9 in IE 7 mode?

**Comment by jdalton** [comment]
> What will this do to, e.g., IE 9 in IE 7 mode?

Support has been removed for IE < 9 and that means older compat modes. Folks using IE9 can [use the Edge compat mode](https://msdn.microsoft.com/en-us/library/ff955275(v=vs.85).aspx) to ensure its not running in compat mode.

**Comment by jdalton** [comment]
Ping

**Comment by savetheclocktower** [comment]
I'll probably get to this sometime over the holidays when I'm bored. Don't worry; I won't forget about it.

**Comment by jdalton** [comment]
Rock, I've got 2 weeks of down time during the holidays to hammer on things too :gift:

**Comment by savetheclocktower** [comment]
OK, I'm cool with landing this. One thing I'll probably do is put back `Element.extend` but assign it to `Prototype.K`; that function is technically part of the public API and I don't want to surprise anyone who's relying on it.

**Comment by savetheclocktower** [comment]
Whoops, you already did that. I am good at reading.

**Comment by jdalton** [comment]
:fireworks: !

**Pull request merged**